adding typescript and some php (#1117)
temporalio/documentation: docs/typescript/activities.md

- Unlike [Workflows](/docs/typescript/determinism), Activities execute in the standard Node.js environment. Any code that needs to talk to the outside world needs to be in an Activity, not a Workflow.
- **Separate from Workflows**: Activities cannot be in the same file as Workflows and must be separately registered (see below for [How to register an Activity on a Worker](#how-to-register-an-activity-on-a-worker))
- **Idempotency**: Activities may be retried repeatedly, so you may need to use [idempotency keys](https://stripe.com/blog/idempotency) for critical side effects.
- The `'@temporalio/activity'` package offers useful utilities for Activity functions such as sleeping, Heartbeating, cancellation, and retrieving metadata (see [docs on Activity Context utilities](#activity-context-utilities) below).

## How to write an Activity Function

> Activities are _just functions_.

Reviewer: ```suggestion
Activities are _simply functions_.
```
Per "just" in https://developers.google.com/style/word-list#letter-j.

});
```

Without Heartbeating, if your activity `StartToCloseTimeout` is 1 hour and the activity stalled or activity worker died, Temporal would have to wait out the 1 hour before retrying.

> But if you used the Heartbeat API, set a `heartbeatTimeout` for 10 seconds, the absence of heartbeats in the `heartbeatTimeout` window would give the Server a Signal that the activity has stalled and should be retried right away rather than at the end of the `StartToCloseTimeout`.

Reviewer: In this case, Temporal Signals aren't involved, right? Perhaps something like the following?
```suggestion
But if you used the Heartbeat API, set a `heartbeatTimeout` for 10 seconds, the absence of Heartbeats in the `heartbeatTimeout` window would inform the Server that the Activity has stalled and should be retried right away rather than at the end of the `StartToCloseTimeout`.
```

> The second major benefit of heartbeating is being able to resume from failure by checkpointing data as `heartbeatDetails`.

Reviewer: ```suggestion
The second major benefit of Heartbeating is being able to resume from failure by checkpointing data as `heartbeatDetails`.
```
